src/clib/pio_internal.h

```diff
@@ -79,7 +79,7 @@ extern "C" {
     int pio_num_iosystem(int *niosysid);
 
     file_desc_t *pio_get_file_from_id(int ncid);
-    int pio_get_file_from_id2(int ncid, file_desc_t **filep);
+    int pio_get_file(int ncid, file_desc_t **filep);
```

Comment:
Why not just replace pio_get_file_from_id() with this new interface (instead of adding a new function - pio_get_file() )?